<p align="center">
  <img src="https://github.com/wejlab/wejlab.github.io/blob/main/wejlab_logo.jpg"  width=50%>
</p>

# Welcome to the Johnson Lab Software Page

## Metascope

![Evan Johnson's Personal GitHub Repository](https://github.com/wevanjohnson)
![WEJ Lab GitHub Repository](https://github.com/wejlab/)

\begin{enumerate}
	\item {\bf ComBat, Combat-Seq, BatchQC:}{\it Combining Multiple Batches of Microarray Genomic Data}. ComBat is a software for reducing batch effects when combining microarray data from different labs, experiments, or hybridization batches, or technology platforms. It utilizes an empirical Bayesian linear modeling approach to robustly account for technical variability across multiple high-thoughput studies. ComBat-Seq uses a Negative Binomial error model to extend ComBat to sequencing-based experiments. BatchQC is a user-interface for interactive evaluation of batch effects in -omic data. These tools are highly cited ($>$8,000 citations), widely-used, and commonly downloaded (107,477 package downloads in 2020; top 3\% of all packages in R/Bioconductor).  
	
	\item {\bf PathoScope, animalcules:} {\it Metagenomic profiling and multi-sample analysis.} PathoScope is a complete bioinformatics framework for the metagenomic analysis of data from clinical or environmental sequencing samples. PathoScope includes modules for reference genome library extraction and indexing, read quality control and alignment, strain identification, and annotation of results. animalcules is new a graphical user interface for multi-sample metagenomic data analysis.
	
	\item {\bf singleCellTK:} {\it Tools and Workflow for Single Cell RNA-seq Analysis.} The singleCellTK is an NCI-funded project to construct a comprehensive and interactive R-software framework for complete data processing and analysis of single cell RNA-sequencing data from heterogeneous tumor samples. We have developed the singleCellTK with an R/Shiny user interface that enables interactive analysis and visualization in the data.
	
	\item {\bf ASSIGN, TBSignatureProfiler, curatedTBData:} {\it Signature scoring, curation, and validation in TB research.}  ASSIGN utilizes Bayesian factor regression model to identify genomic biomarkers for applications in pathway profiling, drug responsiveness, environmental exposure, and infectious disease diagnosis. The TBSignatureProfiler is a compilation of more than 50 signatures of TB outcomes and includes package tools for scoring samples and visualizing the results. The curatedTBData is an effort to compile and harmonize data from more than 45 datasets with more than 4,000 samples. The curatedTBData can be combined with the TBSignatureProfiler to generate or validate new signatures, evaluate existing signatures on subsets or to provide data for other projects.    
	
	\item {\bf SCAN-UPC, MAT, MA2C:} {\it Single and Multi-Channel Array Normalization and barcoding.} Probability Codes. SCAN is a microarray normalization method that removes background noise using only data from within each array individually, therefore facilitating applications in precision medicine. UPC utilizes a similar modeling approach to produce barcodes that estimate gene activity in data from microarray and RNA-sequencing platforms. MAT is designed for the analysis of data from Affymetrix tiling microarrays. The MA2C software is a similar approach but designed to analyze data from two-color tiling arrays.
	
	\item {\bf GNUMAP:} {\it Genomic Next-generation Universal Mapper.} GUNUMAP is a software suite for aligning next sequencing data from DNA-seq, BS-seq, and RNA-seq (including small RNAs, RNA editing) experiments. It uses a highly accurate probabilistic alignment approach that incorporates base uncertainty into the alignment algorithm.

\end{enumerate}

